src/tribol/mesh/MeshData.cpp

```diff
@@ -437,255 +475,111 @@ bool MeshData::computeFaceData( int const dim )
+          // tolerance
+          area[i] += 0.5 * magnitude( nX, nY, nZ );
         }
 
-        // compute the pallet normal contribution for the last pallet
-        auto nodeIndex = m_numNodesPerCell * i;
-        auto nodeId = m_connectivity[ nodeIndex ];
-        auto nextNodeId = m_connectivity[ nodeIndex + m_numNodesPerCell - 1 ];
-        vX1 = m_positionX[ nodeId ] - m_positionX[ nextNodeId ];
-        vY1 = m_positionY[ nodeId ] - m_positionY[ nextNodeId ];
-        vZ1 = m_positionZ[ nodeId ] - m_positionZ[ nextNodeId ];
-
-        vX2 = m_cX[i] - m_positionX[ nextNodeId ];
-        vY2 = m_cY[i] - m_positionY[ nextNodeId ];
-        vZ2 = m_cZ[i] - m_positionZ[ nextNodeId ];
-        
-        nX = (vY1 * vZ2) - (vZ1 * vY2);
-        nY = (vZ1 * vX2) - (vX1 * vZ2);
-        nZ = (vX1 * vY2) - (vY1 * vX2);
-
-        // sum the normal component contributions into the component variables
-        m_nX[i] += nX;
-        m_nY[i] += nY;
-        m_nZ[i] += nZ;
-
-        // half the magnitude of the computed normal is the pallet area. Note: this is exact 
-        // for planar faces and approximate for warped faces. Face areas are used in a general 
-        // sense to create a face-overlap tolerance
-        m_area[i] += 0.5 * magnitude( nX, nY, nZ );
-
         // multiply the pallet normal components by fac to obtain avg.
-        m_nX[i] = fac * m_nX[i];
-        m_nY[i] = fac * m_nY[i];
-        m_nZ[i] = fac * m_nZ[i];
+        n[0][i] = fac * n[0][i];
+        n[1][i] = fac * n[1][i];
+        n[2][i] = fac * n[2][i];
 
         // compute the magnitude of the average pallet normal
-        auto mag = magnitude(m_nX[i], m_nY[i], m_nZ[i] );
-        auto invMag = nrmlMagTol;
-        if (mag >= nrmlMagTol) {
-           invMag = 1.0 / mag;
+        auto mag = magnitude(n[0][i], n[1][i], n[2][i] );
+        auto inv_mag = nrml_mag_tol;
+        if (mag >= nrml_mag_tol) {
+          inv_mag = 1.0 / mag;
         } else {
-           faceDataOk = false;
+          face_data_ok[0] = static_cast<IndexT>(false);
         }
 
         // normalize the average normal
-        m_nX[i] *= invMag;
-        m_nY[i] *= invMag;
-        m_nZ[i] *= invMag;
+        n[0][i] *= inv_mag;
+        n[1][i] *= inv_mag;
+        n[2][i] *= inv_mag;
 
-     } // end if (dim == 3)
+      } // end if (dim == 3)
 
-  } // end cell loop
+  }); // end element loop
 
-  SLIC_WARNING_IF(!faceDataOk, 
-      axom::fmt::format("There are faces with a normal magnitude less than tolerance ({:e}).", nrmlMagTol));
+  ArrayT<IndexT, 1, MemorySpace::Host> face_data_ok_host(face_data_ok_data);
+  SLIC_WARNING_IF(!face_data_ok_host[0], 
+      fmt::format("There are faces with a normal magnitude less than tolerance ({:e}).", nrml_mag_tol));
 
-  return faceDataOk; 
+  return face_data_ok_host[0];
 
 } // end MeshData::computeFaceData()
 
 //------------------------------------------------------------------------------
-real MeshData::computeFaceRadius( int faceId ) 
```

Comment:
I don't see where a new `computeFaceRadius()` is implemented. I searched for `m_face_radius` and don't see where it is actually computed/populated. Please verify the face radius is still being calculated on the mesh data.

Reply:
I inlined it to the `computeFaceData()` method to match the other calcs.  We could make methods for each step: `computeElementNormal()`, `computeElementCentroid()`, etc. and refactor this if it's preferrable.